arch/x86/assign.c

```diff
@@ -166,6 +166,10 @@ alloc_entry(struct vm *vm, enum ptdev_intr_type type)
 	ASSERT(entry, "alloc memory failed");
 	entry->type = type;
 	entry->vm = vm;
+
+	INIT_LIST_HEAD(&entry->softirq_node);
+	INIT_LIST_HEAD(&entry->entry_node);
```

Comment:
entry->entry_node is added in the next code, do we need to INIT here ?

Reply:
Per the standard operating, it better and safe to call INIT_LIST_HEAD before use list head.